### PR TITLE
EVG-12618 make target for code gen

### PR DESCRIPTION
**Pull request by john-m-liu**
This makes the code generator finally usable! This just adds a simple cli command that calls the code gen code with some hardcoded file paths. I had to add a package-level variable to hold a path prefix, because this command and the tests call it with different working directories

### CODE REVIEW

File: cmd/generate-rest-model/generate-rest-model.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+func main() {
+	var (
+		configFile string
+		schemaFile string
+		modelFile  string
+		helperFile string
+	)
+	flag.StringVar(&configFile, "config", "", "path to the config yml file with struct mappings")
+	flag.StringVar(&schemaFile, "schema", "", "path to the graphql schema to use for the output")
+	flag.StringVar(&modelFile, "model", "", "file path to write the generated model code <TODO relative>")
```
Comment (ablack12) [inline cmd/generate-rest-model/generate-rest-model.go:23]: wait sorry is this TODO for you or are we supposed to understand it

File: cmd/generate-rest-model/generate-rest-model.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	flag.StringVar(&helperFile, "helper", "", "file path to write the generated helper functions")
+	flag.Parse()
+
+	configData, err := ioutil.ReadFile(configFile)
+	grip.EmergencyFatal(errors.Wrap(err, "unable to read config file"))
```
Comment (ablack12) [inline cmd/generate-rest-model/generate-rest-model.go:28]: always exciting to see emergency fatal :)

File: cmd/generate-rest-model/generate-rest-model.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	err = yaml.Unmarshal(configData, &gqlConfig)
+	grip.EmergencyFatal(errors.Wrap(err, "unable to parse config file"))
+	schema, err := ioutil.ReadFile(schemaFile)
+	grip.EmergencyFatal(errors.Wrap(err, "unable to read schema file"))
+
+	mapping := model.ModelMapping{}
+	for dbModel, info := range gqlConfig.Models {
+		mapping[dbModel] = info.Model[0]
+	}
+	model.SetGeneratePathPrefix("rest/model")
```
Comment (ablack12) [inline cmd/generate-rest-model/generate-rest-model.go:39]: Will we not always be making this rest/model? Why not just make patchPrefix a constant?

**Review by ablack12 — CHANGES_REQUESTED** [review]
I'm assuming you tested this 👍 but if you do want someone with makefile experience to review then that would not be me

File: makefile [hunk 4]
```diff
@@ -187,6 +187,13 @@ $(buildDir)/go-test-config:cmd/go-test-config/make-config.go
 	GOPATH=$(shell dirname $(shell pwd)) $(gobin) build -o $@ $<
 #end generated config
 
+# generate rest model
+generate-rest-model:$(buildDir)/generate-rest-model
```
Comment (ablack12) [inline makefile:191]: So in terms of using this, is it just make generate-rest-model? Does the user always put their to-be-generated-struct in rest/model/schema/type_mapping.yml then?

File: cmd/generate-rest-model/generate-rest-model.go (hunk 1 above)
Comment (john-m-liu) [reply cmd/generate-rest-model/generate-rest-model.go]: doh that was a reminder to me to update the help text, which I definitely didn't forget...

File: cmd/generate-rest-model/generate-rest-model.go (hunk 3 above)
Comment (john-m-liu) [reply cmd/generate-rest-model/generate-rest-model.go]: it will. Made it a constant

File: makefile (hunk 4 above)
Comment (john-m-liu) [reply makefile:191]: Correct, my intent is that all the generated stuff will be in a single file. So rest/model/schema/type_mapping.yml will bea list of all of our generated models

**Review by ablack12 — APPROVED** [review]
Yay! 💯

**Pull request merged**